brayns/common/scene/Scene.cpp

```diff
@@ -35,10 +35,39 @@ namespace fs = boost::filesystem;
 namespace
 {
 const size_t CACHE_VERSION = 10;
+
+template <typename T, typename U = T> // U seems to be needed when getID is a
+                                      // member function of a base of T.
+std::shared_ptr<T> _find(const std::vector<std::shared_ptr<T>>& list,
```

Comment:
I think it would look nicer to take a lambda instead of a getID method, so you call _find_if(list, [id](const auto& x){ return x==id;}).

Reply:
I must be missing something because what you say doesn't compile in my head.

Reply:
I meant like this: https://gist.github.com/karjonas/b59a568c07f26acafa5f2541557292f8

Reply:
It's more generic, but I don't find it nicer because the client code is longer. I'd change it only if the function needs to be more generic.